### PR TITLE
minor: PHP8.2 - test for DNF types

### PR DESCRIPTION
**Pull request by keradus**

### CODE REVIEW

**Comment by keradus** [comment]
@SpacePossum , @Wirone , @kubawerlos  - maybe any of you can help to make it pass?

**Comment by Wirone** [comment]
@keradus there was #6608 issue and #6629 PR, but these were closed just like `readonly` support. If @SpacePossum agrees, I can resurrect it too and continue the work.

**Comment by SpacePossum** [comment]
I haven't checked anything for a long time so I cannot say what the state is of https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6629 
If you are up for it, please have a look @Wirone :)  but also, feel free to start from scratch, what ever works best for you :+1:

**Comment by keradus** [comment]
Thanks for picking it up!
I believe it's the last topic to claim 8.2 support

File: tests/Fixtures/Integration/misc/PHP8_2.test [hunk 1]
```diff
@@ -88,6 +88,11 @@ trait WithConstants
     private const THREE = 'three';
 }
 
+// https://wiki.php.net/rfc/dnf_types
+function generateSlug((HasTitle&HasId)|null $post) {
+    throw new \Exception("not implemented);
```
Comment (kubawerlos) [inline tests/Fixtures/Integration/misc/PHP8_2.test:93]:
```suggestion
    throw new \Exception("not implemented");
```

File: tests/Fixtures/Integration/misc/PHP8_2.test [hunk 2]
```diff
@@ -165,3 +170,8 @@ trait      WithConstants      {
     protected const TWO   = 'two';
     private   const THREE = 'three';
 }
+
+// https://wiki.php.net/rfc/dnf_types
+function generateSlug((HasTitle&HasId)|null $post) {
+    throw new \Exception("not implemented);
```
Comment (kubawerlos) [inline tests/Fixtures/Integration/misc/PHP8_2.test:176]:
```suggestion
    throw new \Exception("not implemented");
```

File: tests/Fixtures/Integration/misc/PHP8_2.test [hunk 3]
```diff
@@ -88,6 +88,11 @@ trait WithConstants
     private const THREE = 'three';
 }
 
+// https://wiki.php.net/rfc/dnf_types
+function generateSlug((HasTitle&HasId)|null $post) {
+    throw new \Exception("not implemented");
```
Comment (kubawerlos) [inline tests/Fixtures/Integration/misc/PHP8_2.test]:
```suggestion
function generateSlug((HasTitle&HasId)|null $post)
{
    throw new \Exception('not implemented');
```

File: tests/Fixtures/Integration/misc/PHP8_2.test [hunk 4]
```diff
@@ -165,3 +170,8 @@ trait      WithConstants      {
     protected const TWO   = 'two';
     private   const THREE = 'three';
 }
+
+// https://wiki.php.net/rfc/dnf_types
+function generateSlug((HasTitle&HasId)|null $post) {
+    throw new \Exception("not implemented");
```
Comment (kubawerlos) [inline tests/Fixtures/Integration/misc/PHP8_2.test:176]:
```suggestion
// https://wiki.php.net/rfc/dnf_types
function generateSlug((HasTitle&HasId)|null $post)
{
    throw new \Exception('not implemented');
```

**Pull request closed without merging**